x-pack/legacy/plugins/index_management/public/components/template_form/steps/use_json_step.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { isJSON } from '../../../../../../../../src/plugins/es_ui_shared/static/validators/string';
+import { StepProps } from '../types';
+
+interface Parameters {
+  prop: string;
```

Comment:
do you think we should be more restrictive here? e.g., prop: `'aliases' | 'settings' | 'mappings'`?

Reply:
We "should" not but it'd be a nice small safety guard. I'll add it 👍